services/supervisor/src/schema.rs

```diff
@@ -80,4 +85,22 @@ impl Schema<Prefixed<'_, &Fork>> {
         let new_configuration_number = self.configuration_number.get().unwrap_or(0) + 1;
         self.configuration_number.set(new_configuration_number);
     }
+
+    /// Assign unique identifier for an instance.
+    pub(crate) fn assign_instance_id(&mut self) -> InstanceId {
+        let id = self
+            .vacant_instance_id
+            .get()
+            .unwrap_or(MAX_BUILTIN_INSTANCE_ID);
+        self.vacant_instance_id.set(id + 1);
+        id
+    }
+}
+
+/// Creates an index given its name and access object.
+fn construct<'a, T: Access, U: FromAccess<Prefixed<'a, T>>>(
+    access: &Prefixed<'a, T>,
+    index_name: &str,
+) -> U {
+    FromAccess::from_access(access.clone(), index_name.into()).unwrap()
```

Comment:
I wonder if this unwrap is really safe?

Reply:
Error is returned if the index exists but has a different type. Since this function is private and used only in the schema constructor (thus types are correct by design), this unwrap is indeed safe.